docs/docs/why-gatsby-uses-graphql.md

```diff
@@ -0,0 +1,360 @@
+2. The image paths are absolute from the _built_ site, not the source code, which makes it confusing to know how to find them from the JSON.
+3. The images are unoptimized, and any optimization you do would have to be manual.
+4. To create a preview listing of all products, we’d need to pass _all_ of the product info in `context`, which will get unweildy as the number of products increases.
+5. It’s not very obvious where data is coming from in the templates that render the pages, so updating the data might be confusing later.
+
+To overcome these limitations, Gatsby introduces GraphQL as a data management layer.
+
+## Create pages using GraphQL
+
+There’s a bit more up-front setup required to get data into GraphQL, but the benefits often far outweigh the cost.
+
+Using `data/products.json` as an example, by using GraphQL we’re able to solve all of the limitations from the previous section:
+
+1. The images can be colocated with the products in `data/images/`.
+2. Image paths in `data/products.json` can be relative to the JSON file.
+3. Gatsby can automatically optimize images for faster loading and better user experience.
+4. We no longer need to pass all product data into `context` when creating pages.
+5. Data is loaded using GraphQL in the components where it’s used, making it much easier to see where data comes from and how to change it.
+
+### Add the necessary plugins to load data into GraphQL
+
+In order to load the product and image data into GraphQL, we need to add a few [Gatsby plugins](/plugins/). Namely, we need plugins to:
+
+- Load the JSON file into GraphQL ([`gatsby-source-filesystem`](/packages/gatsby-source-filesystem/))
+- Convert JSON files into GraphQL “nodes” ([`gatsby-transformer-json`](/packages/gatsby-transformer-json/))
+- Optimize images ([`gatsby-plugin-sharp`](/packages/gatsby-plugin-sharp/))
+- Convert images into GraphQL nodes ([`gatsby-transformer-sharp`](/packages/gatsby-transformer-sharp/))
+
+In addition to the plugins, we’ll use [`gatsby-image`](/packages/gatsby-image/) to display the optimized images with lazy loading.
+
+Install these packages using the command line:
+
+```bash
+npm install --save gatsby-source-filesystem gatsby-transformer-json gatsby-plugin-sharp gatsby-transformer-sharp gatsby-image
+```
+
+Then add them to `gatsby-config.js`:
+
+```js:title=gatsby-config.js
+module.exports = {
+  plugins: [
+    {
+      resolve: "gatsby-source-filesystem",
+      options: {
+        path: "./data/",
+      },
+    },
+    "gatsby-transformer-json",
+    "gatsby-transformer-sharp",
+    "gatsby-plugin-sharp",
+  ],
+}
+```
+
+To check that this worked, let’s use the GraphQL Playground, which is available during development, by running:
+
+```
+GATSBY_GRAPHQL_IDE=playground yarn develop
+```
+
+> **NOTE:** The `GATSBY_GRAPHQL_IDE=playground` part of this command is optional. Adding it enables the GraphQL Playground instead of GraphiQL, which is an older interface for exploring GraphQL.
```

Comment:
I would probably just have used the default here. It's nice to just document the happy path.

(But also I agree, I so much prefer the playground)

Reply:
Also I'm aware this would require some updates to screenshots and other stuff, so this can totally stay too!

Reply:
(I actually did this because I thought the goal was to make the Playground the default after we’re sure it's not introducing any regressions and I wanted to avoid updating all the screenshots.) 😅

Reply:
I like your style 😎 It's a lot better, in my opinion.